delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/AclsRoutes.scala

```diff
@@ -0,0 +1,275 @@
+    }
+    .mapJsonObject(_.remove("_deprecated"))
+
+  implicit val aclWriteFResponseEncoder: JsonLdEncoder[ResourceF[AclAddress, Unit]] = JsonLdEncoder.compactFromCirce(
+    (v: ResourceF[AclAddress, Unit]) => iriLens.get(v.id),
+    iriContext = contexts.resource
+  )
+
+  implicit val searchEncoder: SearchEncoder[AclResource] = encodeResults(_ => None)
+
+  private def extractAclAddress: Directive1[AclAddress] =
+    extractUnmatchedPath.flatMap {
+      case SingleSlash                                                                                            => provide(AclAddress.Root)
+      case Path.Empty                                                                                             => provide(AclAddress.Root)
+      case Path.Slash(Path.Segment(org, Path.Empty)) if org != any                                                => provide(Organization(Label.unsafe(org)))
+      case Path.Slash(Path.Segment(org, Path.Slash(Path.Segment(proj, Path.Empty)))) if org != any && proj != any =>
+        (for {
+          orgLabel  <- Label(org)
+          projLabel <- Label(proj)
+        } yield Project(orgLabel, projLabel)).fold(
+          err => reject(validationRejection(err.getMessage)),
+          provide(_)
+        )
+      case _                                                                                                      => reject
+
+    }
+
+  private def extractAclAddressFilter: Directive1[AclAddressFilter] =
+    (extractUnmatchedPath & parameter("ancestors" ? false)).tflatMap { case (path, ancestors) =>
+      path match {
+        case Path.Slash(Path.Segment(`any`, Path.Empty))                                  => provide(AnyOrganization(ancestors))
+        case Path.Slash(Path.Segment(`any`, Path.Slash(Path.Segment(`any`, Path.Empty)))) =>
+          provide(AnyOrganizationAnyProject(ancestors))
+        case Path.Slash(Path.Segment(org, Path.Slash(Path.Segment(`any`, Path.Empty))))   =>
+          Label(org).fold(
+            err => reject(validationRejection(err.getMessage)),
+            label => provide(AnyProject(label, ancestors).asInstanceOf[AclAddressFilter])
+          )
+        case _                                                                            => reject
+      }
+    }
+
+  def routes: Route = baseUriPrefix(baseUri.prefix) {
+    pathPrefix("acls") {
+      extractCaller { implicit caller =>
+        concat(
+          extractAclAddress { address =>
+            extractSubject { implicit subject =>
+              parameter("rev" ? 0L) { rev =>
+                concat(
+                  (put & entity(as[AclInput])) { aclEntity =>
+                    authorizeFor(address, aclsWrite).apply {
+                      val status = if (rev == 0L) Created else OK
+                      completeIO(status, acls.replace(address, aclEntity.toAcl, rev).map(_.void))
+                    }
+                  },
+                  (patch & entity(as[PatchAcl])) { aclPatch =>
```

Comment:
it would be cleaner if you combine the current directives with the `authorizeFor` directive:
```scala
(patch & entity(as[PatchAcl]) & authorizeFor(address, aclsWrite)) {
   case AppendAcl(acl) => ???
   case SubtractAcl(acl) => ???
}
```